src/adapters/source-adapters/BoundedUnionSourceAdapter.sol

```diff
@@ -55,6 +55,22 @@ abstract contract BoundedUnionSourceAdapter is
         return _selectBoundedPrice(clAnswer, clTimestamp, crAnswer, crTimestamp, pyAnswer, pyTimestamp);
     }
 
+    /**
+     * @notice Returns the requested round data from the source.
+     * @dev Not all aggregated adapters support this, so this returns uninitialized data.
+     * @return answer Round answer in 18 decimals.
+     * @return updatedAt The timestamp of the answer.
+     */
+    function getSourceDataAtRound(uint256 /* roundId */ )
+        public
+        view
+        virtual
+        override(ChainlinkSourceAdapter, ChronicleMedianSourceAdapter, PythSourceAdapter)
+        returns (int256, uint256)
+    {
+        return (0, 0);
```

Comment:
@Reinis-FRP one question on this. If the round matches the current round (i.e. 1) shouldn't we return the current values?

Reply:
We don't have access to current round at union here, and even don't know if the source that got chosen by union has rounds

Reply:
For the bounded union, specifically, it can't really use rounds, right? Since it may have some with rounds and some without, and the rounds would be inconsistent between sources.

But for sources that don't have rounds, we always return a round of 1, so my thinking was that if someone requests a round, we return the current price data if they requested round 1, otherwise return 0,0.

Thoughts?

Reply:
I would assume clients requesting a particular round data would expect returned data not to change once its available. In case of Oval where we also check data against unlock timestamp this would result in flip-flopping between uninitialized and the latest data depending on when it is being called.